### PR TITLE
Avoid std::stringstream

### PR DESCRIPTION
**Pull request by matthew-reynolds**
Similar to https://github.com/ros-controls/ros2_control/pull/391

We rarely use the `_STREAM` variants of RCLCPP logging, so we might as well just avoid them altogether to avoid the `std::stringstream`.

While I was in there, I also corrected the printf format for `size_t`. We previously had a mix of `%d` and `%u`. For correctness and portability, it should be `%zu`, which is specifically for `size_t`s.

### CODE REVIEW

File: diff_drive_controller/src/diff_drive_controller.cpp
Comment (destogl) [inline diff_drive_controller/src/diff_drive_controller.cpp:578]:
```suggestion
    RCLCPP_ERROR(logger, "No '%s' wheel names specified", side.c_str());
```

File: forward_command_controller/src/forward_command_controller.cpp
Comment (destogl) [inline forward_command_controller/src/forward_command_controller.cpp:85]: When you are already editing this, can you take a look at the following use?

```suggestion
  RCLCPP_INFO(node_->get_logger(), "configure successful");
```

File: forward_command_controller/src/forward_command_controller.cpp
Comment (destogl) [inline forward_command_controller/src/forward_command_controller.cpp:166]:
```suggestion
      node_->get_logger(),
```

**Review by destogl — APPROVED** [review]
Just a few small suggestions. Let me know what do you think about them. Thanks!

**Comment by bmagyar** [comment]
I am wondering if there's a way to use logging properly without the bloat of `get_logger()` and other ugly stuff like that. `RCUTILS_LOG_ERROR_NAMED` seems to have a similar effect... any opinions/experience with this?

**Review by bmagyar — APPROVED** [review]

**Comment by matthew-reynolds** [comment]
As I recall, the `RCLCPP` logging macros just take a string name from the `Node`'s logger which is then passed down to the `RCUTILS` macros. The `RCUTILS` macros are basically the same as the ROS1 `LOG_NAMED` macros.

I think the cleanest and most standard is to stick with the `RCLCPP` macros, and to use `node_` everywhere. We mostly use `node_` internally and `get_node()` externally, which I think is the right move - It's not too verbose and it matches the standard ROS C++ approach. I've got a PR ready that I'll put up once I get approval from work...

**Pull request merged**
